misc/python/materialize/util.py

```diff
@@ -23,10 +24,30 @@ def nonce(digits: int) -> str:
     return "".join(random.choice("0123456789abcdef") for _ in range(digits))
 
 
-def known_materialize_versions() -> List[Version]:
-    """Returns all known Materialize versions.
+def released_materialize_versions() -> List[Version]:
+    """Returns all Materialize versions except those explicitly marked as
+    unrelased in the metadata of their .md files.
 
     The list is returned in version order with newest versions first.
     """
-    files = Path(ROOT / "doc" / "user" / "content" / "releases").glob("*.md")
-    return [Version.parse(f.stem.lstrip("v")) for f in files if f.stem.startswith("v")]
+    files = Path(ROOT / "doc" / "user" / "content" / "releases").glob("v*.md")
+    versions = []
+    for f in files:
+        with open(f, "r", encoding="utf-8") as input_file:
+            md = markdown.Markdown(extensions=["meta"])
+            md.convert(input_file.read())
+            # Meta is an extension to Markdown and this confuses mypy
+            meta = md.Meta  # type: ignore[attr-defined]
```

Comment:
So this Markdown library is parsing a different kind of Markdown metadata than is actually in use here. This is "frontmatter", which is much more ergonomically parsed with e.g. https://pypi.org/project/python-frontmatter/. I'm happy to take this over since I figure you are probably sick of this by now. :D

Reply:
Thank you for taking this over the line!